### DISCUSSION TITLE
Dropwizard JDBI3 Database Timezone

### DISCUSSION THREAD

**Discussion by Tylerremmie** [dropwizard/dropwizard#4771] [General]
Hello! 
I recently began upgrading my java project to use JDBI3 with Dropwizard and I ran across an issue that hopefully I can get cleared up. In the previous version of Dropwizard jdbi, I was able to override the database timezone when creating a new DBIFactory like this:

`    DBIFactory createDBIFactory()
    {
        return new DBIFactory() {
            @Override
            protected Optional<TimeZone> databaseTimeZone() {
                return Optional.of(TimeZone.getTimeZone("UTC"));
            }
        };
    }
`

Which passed in a timezone to the InstantArgumentFactory to handle InstantArguments to a specify a timezone conversion.

With this I was able to have my JVM set to one timezone, but my database entries were submit as another (UTC). I was doing some testing with my current changes to use the new JDBI factory:
`JdbiFactory createJdbiFactory()
    {
        return new JdbiFactory();
    }`
and I noticed Instant() timestamps were incorrect, which lead me to this issue.

Is there a new implementation or method that can be used to achieve this same result?

Thanks!

**Comment by joschi** [comment]
Using/overriding the time zone for `java.time` (JSR 310) is not supported in Jdbi 3.x.

* https://github.com/jdbi/jdbi/issues/1539
* https://github.com/jdbi/jdbi/issues/1271
* https://github.com/jdbi/jdbi/issues/988

You can keep using JDBI 2.x (via https://github.com/dropwizard/dropwizard-jdbi/) but there won't be any updates for it because the support for JDBI 2.x ended.

The probably equally unsatisfying way to fix this correctly is running your application and the database with the same time zone settings, i. e. UTC.

**Comment by Tylerremmie** [comment]
Alright, I appreciate your input. I would prefer to move ahead with the JDBI3 upgrade since JDBI2 is no longer supported, and I guess I will just run the application in UTC.
